sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionUtils.java

```diff
@@ -0,0 +1,144 @@
+    // Since this is only Java 9+ functionality it needs to use reflection.
+    private static final String MODULE_CLASS = "java.lang.Module";
+    private static final Object CORE_MODULE;
+    private static final MethodHandle GET_MODULE;
+    private static final MethodHandle IS_MODULE_EXPORTED;
+    private static final MethodHandle CAN_READ_MODULE;
+    private static final MethodHandle ADD_MODULE_READ;
+
+    static {
+        MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
+
+        Object coreModule = null;
+        MethodHandle getModule = null;
+        MethodHandle isModuleExported = null;
+        MethodHandle canReadModule = null;
+        MethodHandle addReadModule = null;
+
+        try {
+            Class<?> moduleClass = Class.forName(MODULE_CLASS);
+            Class<?> classClass = Class.class;
+            getModule = publicLookup.findVirtual(classClass, "getModule", methodType(moduleClass));
+            coreModule = getModule.invoke(ReflectionUtils.class);
+            isModuleExported = publicLookup.findVirtual(moduleClass, "isExported",
+                methodType(boolean.class, String.class));
+            canReadModule = publicLookup.findVirtual(moduleClass, "canRead", methodType(boolean.class, moduleClass));
+            addReadModule = MethodHandles.lookup()
+                .findVirtual(moduleClass, "addReads", methodType(moduleClass, moduleClass));
+        } catch (Throwable ex) {
+            new ClientLogger(ReflectionUtils.class)
+                .verbose("Failed to retrieve MethodHandles used to check module information. "
+                    + "If the application is not using Java 9+ runtime behavior will work as expected.", ex);
+        }
+
+        CORE_MODULE = coreModule;
+        GET_MODULE = getModule;
+        IS_MODULE_EXPORTED = isModuleExported;
+        CAN_READ_MODULE = canReadModule;
+        ADD_MODULE_READ = addReadModule;
+    }
+
+    /**
+     * Gets the {@link MethodHandles.Lookup} to use when performing reflective operations.
+     * <p>
+     * If Java 8 is being used this will always return {@link MethodHandles.Lookup#publicLookup()} as Java 8 doesn't
+     * have module boundaries that will prevent reflective access to the {@code targetClass}.
+     * <p>
+     * If Java 9 or above is being used this will return a {@link MethodHandles.Lookup} based on whether the module
+     * containing the {@code targetClass} exports the package containing the class. Otherwise, the {@link
+     * MethodHandles.Lookup} associated to {@code com.azure.core} will attempt to read the module containing {@code
+     * targetClass}.
+     *
+     * @param targetClass The {@link Class} that will need to be reflectively accessed.
+     * @return The {@link MethodHandles.Lookup} that will allow {@code com.azure.core} to access the {@code targetClass}
+     * reflectively.
+     */
+    public static MethodHandles.Lookup getLookupToUse(Class<?> targetClass) {
```

Comment:
Check this with `Throwable` and force callers to handle the exception appropriately, either swallow and use a fallback path or propagate.